README.md

```diff
@@ -1 +1,116 @@
-# check-commit-hook
\ No newline at end of file
+# check-commit-hook
+
+Based on [1] this repository contains a set of BW pre-commit hooks to be run on every git commit, the same hooks are used by the CI/CD pipeline to check the files.
+
+# Hooks
+
+## Checkpatch
+
+Uses Linux kernel checkpatch to check for coding style errors in C code, Makefiles, etc. It uses a custom configuration file to ignore some errors and enforce others.
+
+### Custom configuration file
+
+The script uses a default configuration file located in `bw_check_commit_hook/data/checkpatch.yaml` but it can be overriden by a custom configuration file and passed to the script using the `--config-file` option.
+
+```
+checkpatch --config-file <config_file> <file>
+```
+
+The configuration file is a YAML file with the following structure:
+
+```yaml
+DIR_CONFIGS:
+  <directory>:
+    errors_enabled:
+      - <checkpatch option>
+      - <checkpatch option>
+      - ...
+    errors_ignored:
+      - <checkpatch option>
+      - <checkpatch option>
+    max_line_length: <number>
+```
+
+#### Magic keys
+
+The configuration file supports some magic keys that make it easier to configure the script.
+
+##### `all`
+
+The `all` key can be used to apply the same configuration to all the files in the repository.
+
+```yaml
+all:
+  errors_enabled:
+    - <checkpatch option>
+    - <checkpatch option>
+    - ...
+  errors_ignored:
+    - <checkpatch option>
+    - <checkpatch option>
+  max_line_length: <number>
+```
+
+##### `ERRORS_COMMON`
+
+The `ERRORS_ENABLED` key can be used as a placeholder for all the errors enabled for all the files, and this needs to be part of `errors_enabled` list to be used.
```

Comment:
"all the files" is maybe misleading since it is only be selectively applied. Really the original script had quite a strange/specific way of sharing config between dirs (called COMMON but not always actually common). What do you think? Might it be better to encourage use of YAML anchors? Then `pyyaml` does the work and it is a clearer and documented feature of the config file. The user can alternatively just write out all the config options for each directory in full.

Reply:
That was my first implementation, but anchors weren't' extending in to a single list they show as nested lists and I didn't want to change the core logic too much, so, abandoned the approach.

Reply:
To potentially be investigated in the future.